src/openforms/payments/contrib/worldline/plugin.py

```diff
@@ -0,0 +1,255 @@
+    CheckoutInput,
+    Order,
+    PaymentOptions,
+)
+from .utils import get_payment_status
+
+logger = structlog.stdlib.get_logger(__name__)
+
+
+class WorldlineOptionsSerializer(JsonSchemaSerializerMixin, serializers.Serializer):
+    merchant = PrimaryKeyRelatedAsChoicesField(
+        queryset=WorldlineMerchant.objects.all(),
+        required=True,
+        help_text=_("Merchant to use"),
+    )
+
+
+def _generate_order(payment: SubmissionPayment) -> Order:
+    amount_of_money = AmountOfMoney(
+        currencyCode="EUR", amount=int((payment.amount * 100).to_integral_exact())
+    )
+    return Order(amountOfMoney=amount_of_money)
+
+
+def _generate_checkout_input(
+    request: HttpRequest,
+    payment: SubmissionPayment,
+    payment_plugin: "WorldlinePaymentPlugin",
+) -> CheckoutInput:
+    return_url = payment_plugin.get_return_url(request, payment)
+    return CheckoutInput(returnUrl=return_url)
+
+
+@register("worldline")
+class WorldlinePaymentPlugin(BasePlugin[PaymentOptions]):
+    verbose_name = _("Wordline")
+    configuration_options = WorldlineOptionsSerializer
+
+    def start_payment(
+        self,
+        request: HttpRequest,
+        payment: SubmissionPayment,
+        options: PaymentOptions,
+    ) -> PaymentInfo:
+        order = _generate_order(payment)
+        checkout_input = _generate_checkout_input(request, payment, self)
+        client = options["merchant"].get_checkout_client()
+
+        # See the hostedCheckoutSpecificInput field on
+        # https://docs.direct.worldline-solutions.com/en/api-reference#tag/HostedCheckout/operation/CreateHostedCheckoutApi
+        # for more configurable checkout options.
+        checkout_request = CreateHostedCheckoutRequest()
+        checkout_request.from_dictionary(
+            {
+                "returnCancelState": True,
+                "hostedCheckoutSpecificInput": checkout_input,
+                "order": order,
+            }
+        )
+
+        try:
+            checkout_response = client.create_hosted_checkout(checkout_request)
+        except (ApiException, CommunicationException) as e:
+            payment.status = PaymentStatus.failed
+            payment.save(update_fields=("status",))
+
+            raise APIException(
+                detail="Failed to retrieve redirect URL from payment provider"
+            ) from e
+
+        plugin_options = payment.plugin_options
+        payment.plugin_options = {
+            **plugin_options,
+            "returnmac": checkout_response.returnmac,
+            "checkout_id": checkout_response.hosted_checkout_id,
+        }
+
+        payment.save(update_fields=("plugin_options",))
+
+        return PaymentInfo(
+            type=PaymentRequestType.get,
+            url=checkout_response.redirect_url or "",  # valid for three hours
+            data={},
+        )
+
+    def apply_status(
+        self,
+        payment: SubmissionPayment,
+        worldline_status: str,
+        checkout_status: str,
```

Comment:
shouldn't this be annotated with the constants/enums?

Reply:
This method (currently) gets the plain statuses retrieved with the SDK, they are [converted](https://github.com/open-formulieren/open-forms/blob/0c4446d8e022d29949c2fab263fc048ebd2b9215/src/openforms/payments/contrib/worldline/plugin.py#L146) though inside the method itself.